src/rws_interface.cpp

```diff
@@ -225,6 +225,97 @@ std::vector<cfg::sys::PresentOption> RWSInterface::getCFGPresentOptions()
+        if(single.use_single_type.empty()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "use_joint"))
+      {
+        single.use_joint = xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE);
+        if(single.use_joint.empty()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "base_frame_pos_x"))
+      {
+        std::stringstream ss(xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE));
+        ss >> single.base_frame.pos.x.value;
+        if(ss.fail()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+        single.base_frame.pos.x.value *= 1000.0;
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "base_frame_pos_y"))
+      {
+        std::stringstream ss(xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE));
+        ss >> single.base_frame.pos.y.value;
+        if(ss.fail()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+        single.base_frame.pos.y.value *= 1000.0;
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "base_frame_pos_z"))
+      {
+        std::stringstream ss(xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE));
+        ss >> single.base_frame.pos.z.value;
+        if(ss.fail()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+        single.base_frame.pos.z.value *= 1000.0;
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "base_frame_orient_u0"))
+      {
+        std::stringstream ss(xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE));
+        ss >> single.base_frame.rot.q1.value;
+        if(ss.fail()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "base_frame_orient_u1"))
+      {
+        std::stringstream ss(xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE));
+        ss >> single.base_frame.rot.q2.value;
+        if(ss.fail()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "base_frame_orient_u2"))
+      {
+        std::stringstream ss(xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE));
+        ss >> single.base_frame.rot.q3.value;
+        if(ss.fail()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "base_frame_orient_u3"))
+      {
+        std::stringstream ss(xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE));
+        ss >> single.base_frame.rot.q4.value;
+        if(ss.fail()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "base_frame_coordinated"))
+      {
+        single.base_frame_coordinated = xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE);
```

Comment:
Seeing as all the other elements `throw` when they are empty: please add a comment explaining *why* this is allowed to be empty. Just so we remember in a few weeks/months time (and for new users), and this doesn't look like we forgot to add a check here.

Reply:
Fixed in 7fb75a6